### ISSUE TITLE
`nonminimal_bool` complains about asserts in boolean expressions

### ISSUE THREAD

**Issue by TomFryersMidsummer** [rust-lang/rust-clippy#11932]
### Summary

`nonminimal_bool` seems to get confused when `assert` and `debug_assert` crop up in boolean expressions: see the example. The issue disappears if `x % 2 == 0 ||` is removed.

It's possible I'm missing something, here, and there is a simpler way to write this. In that case, it'd be nice if the error message was a bit clearer what this was.

### Lint Name

nonminimal_bool

### Reproducer

I tried this code:

```rust
fn f(x: i32) -> bool {
    x % 2 == 0 || {
        assert!(x > 0);
        x % 3 == 0
    }
}
```

I saw this happen:

```
warning: this boolean expression can be simplified
 --> foo.rs:3:9
  |
3 |         assert!(x > 0);
  |         ^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
  = note: `#[warn(clippy::nonminimal_bool)]` on by default
  = note: this warning originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen: [no error]

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```

**Comment by asquared31415** [comment]
That code expands to
```rust
fn f(x: i32) -> bool {
    x % 2 == 0 ||
        {
            if !(x > 0) {
                    ::core::panicking::panic("assertion failed: x > 0")
                };
            x % 3 == 0
        }
}
```
and what clippy is seeing is that `if !(x > 0)`, which could indeed be simplified.
If you run the expanded code through clippy, that is indeed where it's warning:
```rust
warning: this boolean expression can be simplified
  --> src/lib.rs:10:16
   |
10 |             if !(x > 0) {
   |                ^^^^^^^^ help: try: `x <= 0`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
   = note: `#[warn(clippy::nonminimal_bool)]` on by default
```

I think that it would be worth clippy special casing the `assert` family of macros to ignore them, because they are always constructed in this way that will *never* be minimal, to simplify the implementation of the macro.
